### PR TITLE
[DDW-105] Force setting password on passwordless wallets

### PR DESCRIPTION
**Pull request by DeeJayElly**
This PR forces setting password on password less wallets by showing the dialog which informs the user that needs to set the password for his wallet.

## Todos

- [x] Fix Storybook wallet context handling (@tomothespian)
- [x] Force creating wallet address on force-password-set action: [DDW-125](https://jira.iohk.io/browse/DDW-125) (@tomothespian)

## Screenshots

Screenshot 2020-04-09 09 13 04
Screenshot 2020-04-09 09 12 37

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1586443857354800)

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by DeeJayElly** [comment]
> @DeeJayElly this is not a good direction. Let's discuss this once you start today.
> The implementation needs to be similar to this: https://github.com/input-output-hk/daedalus/blob/chore/ddw-105-force-setting-password-on-passwordless-wallets/source/renderer/app/components/wallet/not-responding/NotResponding.js

Implemented @nikolaglumac

**Comment by a-rukin** [comment]
image
shouldn't we show summary part behind this overlay? Or we just can't?

**Comment by a-rukin** [comment]
image
this dialog now should be dissmissable

**Comment by DeeJayElly** [comment]
> image
> 
> this dialog now should be dissmissable

Added

**Comment by DeeJayElly** [comment]
> image
> 
> shouldn't we show summary part behind this overlay? Or we just can't?

You mean like transparent overlay?

**Comment by nikolaglumac** [comment]
Yes it should be an overlay with transparent background

**Comment by DeeJayElly** [comment]
> Yes it should be an overlay with transparent background

@a-rukin . @nikolaglumac Now it is.

Screenshot 2020-04-09 15 57 56

**Comment by DeeJayElly** [comment]
> > Yes it should be an overlay with transparent background
> 
> @a-rukin . @nikolaglumac Now it is.
>

**Comment by DeeJayElly** [comment]
> image
> 
> shouldn't we show summary part behind this overlay? Or we just can't?

Screenshot 2020-04-09 16 05 29

**Comment by gnpf** [comment]
@DeeJayElly I just checked locally and the background is not transparent on the theme "Incentivized Testnet"

image

**Comment by DeeJayElly** [comment]
> @DeeJayElly I just checked locally and the background is not transparent on the theme "Incentivized Testnet"
> 
> image

Fixed

File: storybook/stories/_support/utils.js [hunk 1]
```diff
@@ -44,7 +45,7 @@ export const generateWallet = (
     reward: new BigNumber(reward).dividedBy(LOVELACES_PER_ADA),
     createdAt: new Date(),
     name,
-    hasPassword: false,
+    hasPassword: hasPassword || true,
```
Comment (nikolaglumac) [inline storybook/stories/_support/utils.js:48]: @DeeJayElly this line will always make `hasPassword` to be `true` as even if you pass in `hasPassword` as false `hasPassword || true` will turn it to `true`.

**Comment by nikolaglumac** [comment]
@DeeJayElly please fix the flow issue 🙏

File: storybook/stories/_support/utils.js (hunk 1 above)
Comment (DeeJayElly) [reply storybook/stories/_support/utils.js]: Sorry, its a leftover i forgot to revert back this part. Its fixed now.

**Comment by DeeJayElly** [comment]
> @DeeJayElly I have found two issues with storybook:
> 
> 1. The general context handling of wallets is wrong and when I change wallets on the "Wallet Summary" story I don't see the wallet balances getting updated nor does the "Set password" overlay gets shown for the wallets with no password set.
> 
> Screenshot 2020-04-14 at 09 22 08
> 
> 1. The "Set password" story shows the "Change wallet password" instead of "Set wallet password" dialog once you update the knob...
> 
> Screenshot 2020-04-14 at 09 21 57

2. Fixed

**Comment by nikolaglumac** [comment]
@DeeJayElly @daniloprates I have asked @tomothespian to fix the two remaining Todos:
Screenshot 2020-04-15 at 09 36 53

**Pull request merged**